code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -97,6 +97,13 @@
 	if(!on)
 		return 0
 
+	// Coolers don't heat.
+	if(air_contents.temperature <= target_temperature && cooling)
+		return 0
```

Comment:
the return values mean nothing here, please remove them from your added code (and the surrounding code also if you would like)

Reply:
Of course they don't. Why would I expect existing code to be sensible in its returns?